File: tools/sap/src/main/java/top/bettercode/summer/tools/sap/connection/DestinationDataProviderImpl.java
package top.bettercode.summer.tools.sap.connection;

import com.sap.conn.jco.ext.DestinationDataEventListener;
import com.sap.conn.jco.ext.DestinationDataProvider;
import java.util.HashMap;
import java.util.Map;
import java.util.Properties;
import org.springframework.util.StringUtils;

public class DestinationDataProviderImpl implements DestinationDataProvider {

  private final Map<String, Properties> provider = new HashMap<>();

  public void addDestinationProperties(String destName, Properties props) {
    provider.put(destName, props);
  }

  @Override
  public Properties getDestinationProperties(String destName) {
    if (!StringUtils.hasText(destName)) {
      throw new NullPointerException("Destinantion name is empty.");
    }

    if (provider.size() == 0) {
      throw new IllegalStateException("Data provider is empty.");
    }

    return provider.get(destName);
  }

  @Override
  public void setDestinationDataEventListener(DestinationDataEventListener listener) {
    throw new UnsupportedOperationException();
  }

  @Override
  public boolean supportsEvents() {
    return false;
  }
}

File: tools/sap/src/main/kotlin/top/bettercode/summer/tools/sap/connection/DestinationDataProviderImpl.kt
package top.bettercode.summer.tools.sap.connection

import com.sap.conn.jco.ext.DestinationDataEventListener
import com.sap.conn.jco.ext.DestinationDataProvider
import org.springframework.util.StringUtils
import java.util.*

class DestinationDataProviderImpl : DestinationDataProvider {
    private val provider: MutableMap<String, Properties> = HashMap()
    fun addDestinationProperties(destName: String, props: Properties) {
        provider[destName] = props
    }

    override fun getDestinationProperties(destName: String): Properties {
        if (!StringUtils.hasText(destName)) {
            throw NullPointerException("Destinantion name is empty.")
        }
        check(provider.isNotEmpty()) { "Data provider is empty." }
        return provider[destName]!!
    }

    override fun setDestinationDataEventListener(listener: DestinationDataEventListener) {
        throw UnsupportedOperationException()
    }

    override fun supportsEvents(): Boolean {
        return false
    }
}